SU2_CFD/src/solvers/CTurbSASolver.cpp

```diff
@@ -289,6 +286,24 @@ void CTurbSASolver::Postprocessing(CGeometry *geometry, CSolver **solver_contain
   AD::EndNoSharedReading();
 }
 
+void CTurbSASolver::Viscous_Residual(unsigned long iEdge, CGeometry* geometry, CSolver** solver_container,
+                                     CNumerics* numerics, CConfig* config) {
+
+  /*--- Define an object to set solver specific numerics contribution. ---*/
+  struct SolverSpecificNumerics {
+
+    FORCEINLINE void operator() (const CGeometry& geometry, CNumerics& numerics, CConfig& config, const CScalarVariable& nodes, unsigned long iPoint, unsigned long jPoint) const {
+      /*--- Roughness heights. ---*/
+      if (config.GetKind_Turb_Model() == SA)
+        numerics.SetRoughness(geometry.nodes->GetRoughnessHeight(iPoint), geometry.nodes->GetRoughnessHeight(jPoint));
+    }
```

Comment:
I know this is probably redundant, but I kept it for know to see whether the regression test pass. (this commit shouldn't change anything logically)
Is there any reason why surface roughness and other SA-variants should not work together

Reply:
It is not implemented for other types of SA source, unfortunately.
Is there an incompatibility @koodlyakshay ?
The work of @suargi might fix this.

Reply:
I know the roughness model works with the noft2 version (which we have in SU2 now) and the standard version of SA. I have not seen it used with other versions anywhere.

Reply:
Ok thanks, so the condition `if SA` can stay 👍